src/Shared/EscapingUtilities.cs

```diff
@@ -214,28 +226,26 @@ string unescapedString
         /// </summary>
         /// <param name="escapedString"></param>
         /// <returns></returns>
-        internal static bool ContainsEscapedWildcards
-            (
-            string escapedString
-            )
+        internal static bool ContainsEscapedWildcards(string escapedString)
         {
-            if (-1 != escapedString.IndexOf('%'))
+            if (escapedString.Length < 3)
             {
-                // It has a '%' sign.  We have promise.
-                if (
-                        (-1 != escapedString.IndexOf("%2", StringComparison.Ordinal)) ||
-                        (-1 != escapedString.IndexOf("%3", StringComparison.Ordinal))
-                    )
+                return false;
+            }
+            int index = escapedString.IndexOf('%', 0, escapedString.Length - 2);
```

Comment:
Do we have good test coverage here?

Reply:
We certainly have [some](https://github.com/dotnet/msbuild/blob/0def9d13d74d075e62851630fe296bb019e401a1/src/Shared/UnitTests/EscapingUtilities_Tests.cs#L74-L85), though it wouldn't be bad to add a couple new cases.

Reply:
I have added a few more cases, thank you!